### DISCUSSION TITLE
adding chess.pgn.Game to organize your boar and moves together for easier pgn management

### DISCUSSION THREAD

**Discussion by nicvagn** [HarshilPatel007/YACS#3] [Ideas]
https://python-chess.readthedocs.io/en/latest/_modules/chess/pgn.html#GameBuilder.result

If you organized your moves and board within a python chess game, It would be a lot easier to display the pgn. I am trying to add pgn parsing/traversal to YACS, but I am having to create chess.pgn.Games post hawk. It would be a lot simpler if when you created the board, you created a Game instead, and used the Games built in board. It will make it much easier in the long run, if you want to traverse pgn.

The alternative is reinventing a tone of stuff already available in python chess. If in stead of creating a board, you create a game I think things will be smother.

Cheers,

nrv

**Comment by nicvagn** [comment]
I am a real command line type of programmer, so qt is throwing me for a loop sorry

**Reply by HarshilPatel007** [reply]
Hi @nicvagn ,
Thanks for taking interest and helping this project.
Even I'm new to QT & python-chess as well. This is my very first application with QT. I'm doing with the help of internet. (of course 😄)

> It would be a lot simpler if when you created the board, you created a Game instead, and used the Games built in board.

> If in stead of creating a board, you create a game I think things will be smother.

To my knowledge, it is necessary to create a `Board()` in order to move forward.
What we can do is, store moves in a list after (or before to store as a san format) pushing a move to the board and do something like this,
```python
import io

pgn = io.StringIO(move_list)
game = chess.pgn.read_game(pgn)
```

to read the game.

**Reply by HarshilPatel007** [reply]
I've already created a POC program before and currently (re)doing it as a full featured project.
The POC code is a single python file and it is a mess. So, I'm re-structuring the code with separating code logic with appropriate files (or modules if necessary) along with the tons of refactoring from what it is.


> If you organized your moves and board within a python chess game, It would be a lot easier to display the pgn. I am trying to add pgn parsing/traversal to YACS, but I am having to create chess.pgn.Games post hawk. It would be a lot simpler if when you created the board, you created a Game instead, and used the Games built in board. It will make it much easier in the long run, if you want to traverse pgn.

If you're already working on saving pgn then I'll not push my code (pgn & displaying moves related because, I've it already with POC code) to main branch. Currently I'm working on a core chessboard features. (not related to pgn. mostly related to moving a piece, promoting, castling, highlighting legal moves, highlighting source - destination squares after a move, marking a squares, drawing an arrows, etc.).
Feel free to make a PR if you want. I'll wait by then.

**Reply by nicvagn** [reply]
If you have the code already, push it. I would be more comfortable refactoring than starting from scratch.


I discovered you can't add moves to games after posting that. You should be able to do that. Ffs

**Reply by HarshilPatel007** [reply]
@nicvagn , I also discovered that thing while coding. It's odd but, this is how python-chess works. 😄

Alright, please check your email or GitHub account for the notification.

Update : did you get the notification?

**Reply by nicvagn** [reply]
yes
